hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -717,13 +702,25 @@ func (r *HostedClusterReconciler) reconcile(ctx context.Context, req ctrl.Reques
 	// conditions (so that it could incorporate e.g. HostedControlPlane and IgnitionServer
 	// availability in the ultimate HostedCluster Available condition)
 	{
-		meta.SetStatusCondition(&hcluster.Status.Conditions, computeHostedClusterAvailability(hcluster, hcp))
+		availableCondition := computeHostedClusterAvailability(hcluster, hcp)
+		_, isHasBeenAvailableAnnotationSet := hcluster.Annotations[hcmetrics.HasBeenAvailableAnnotation]
 
-		// SLI: Hosted Cluster available duration.
-		reportAvailableTime(hcluster)
-	}
+		meta.SetStatusCondition(&hcluster.Status.Conditions, availableCondition)
+
+		if availableCondition.Status == metav1.ConditionTrue && !isHasBeenAvailableAnnotationSet {
+			original := hcluster.DeepCopy()
+
+			if hcluster.Annotations == nil {
+				hcluster.Annotations = make(map[string]string)
+			}
 
-	reportClusterVersionRolloutTime(hcluster)
+			hcluster.Annotations[hcmetrics.HasBeenAvailableAnnotation] = "true"
+
+			if err := r.Patch(ctx, hcluster, client.MergeFromWithOptions(original)); err != nil {
```

Comment:
why is this patch needed here now vs before?

Reply:
Hello Alberto,
There was a bug in the old code: the `hostedcluster` object was annotated in memory but the annotation was not saved in DB.
I believe this is due to this code which only saves the status change but nothing else: 
https://github.com/openshift/hypershift/blob/main/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L322
Here is some other place in the code in which the same kind of call to `patch` in needed to save annotations:
https://github.com/openshift/hypershift/blob/7fb56f7c9af2f27d207e1d68d4b8fc824cc2b798/hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go#L3405